plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -100,11 +97,6 @@ def __eq__(self, other):
         :return:      True if self and other are equal, False otherwise
         :rtype:       bool
         """
-
-        # Simple case, they are entirely different
-        if self.name != other.name:
```

Comment:
This suffers from the same problem above. ``kernel-4.5.7-200`` and ``kernel-devel-4.5.7-200`` would compare as equal.